File: frontend/lib/api/orchestration.ts
import { apiClient } from "@/lib/api/client";
import type { UiContextPayload } from "@/lib/api/integrated-ui";

export async function runOrchestration(
  workflow: string,
  context: UiContextPayload,
  inputPayload: Record<string, any> = {}
) {
  return apiClient.post<any>("/orchestration/run", {
    workflow,
    persona: context.persona,
    scope_type: context.scope_type,
    scope_id: context.scope_id,
    period: context.period,
    compare_to: context.compare_to,
    input_payload: inputPayload
  });
}

File: app/api/routers/orchestration.py
from __future__ import annotations

from fastapi import APIRouter
from pydantic import BaseModel, Field

from app.orchestration import get_orchestration_engine
from app.shared.responses import ok

router = APIRouter(prefix="/orchestration", tags=["Orchestration"])


class OrchestrationRequest(BaseModel):
    workflow: str = Field(default="dashboard")
    persona: str = "Advisor"
    scope_type: str = "Advisor"
    scope_id: str = "ADV0001"
    period: str = "YTD"
    compare_to: str = "Prior Year"
    input_payload: dict = Field(default_factory=dict)


@router.post("/run")
def run_orchestration(request: OrchestrationRequest):
    context = {
        "persona": request.persona,
        "scope_type": request.scope_type,
        "scope_id": request.scope_id,
        "period": request.period,
        "compare_to": request.compare_to,
    }
    result = get_orchestration_engine().run(
        workflow=request.workflow,
        context=context,
        input_payload=request.input_payload,
    )
    return ok(data=result)
